tests/test_js.py

```diff
@@ -80,6 +80,49 @@ async def test_publish_verbose(self):
 
         await nc.close()
 
+    @async_test
+    async def test_publish_async(self):
+        nc = NATS()
+        await nc.connect()
+        js = nc.jetstream()
+
+        future = await js.publish_async("foo", b'bar')
+        with pytest.raises(NoStreamResponseError):
+            await future
+
+        await js.add_stream(name="QUUX", subjects=["quux"])
+
+        futures = [
+            await js.publish_async("quux", b'bar:1') for i in range(0, 100)
+        ]
+        results = await asyncio.gather(*futures)
+
+        for seq, result in enumerate(results, 1):
+            self.assertEqual(result.stream, "QUUX")
+            self.assertEqual(result.seq, seq)
+
+        # Check that stalled is raised and can be recovered
+        futures = []
+        while True:
+            try:
+               future = await js.publish_async("quux", b'bar:1')
+               futures.append(future)
+            except TooManyStalledMsgsError:
```

Comment:
In the go client there is also a publish_async_complete to wait when the jetstream context is not stalled anymore, we could do something similar here (could be a different PR)